sycl/source/kernel_bundle.cpp

```diff
@@ -321,7 +322,18 @@ std::vector<sycl::device> find_device_intersection(
 //////////////////////////
 
 std::vector<kernel_id> get_kernel_ids() {
-  return detail::ProgramManager::getInstance().getAllSYCLKernelIDs();
+  std::vector<kernel_id> ids =
+      detail::ProgramManager::getInstance().getAllSYCLKernelIDs();
+  // Filter out kernel ids coming from RTC kernels in order to be
+  // spec-compliant. Kernel ids from RTC are prefixed with rtc_NUM$, so looking
+  // for '$' should be enough.
+  ids.erase(std::remove_if(ids.begin(), ids.end(),
```

Comment:
@cperkinsintel LGTM but I appreciate if you can take a look.

Reply:
But this would remove kernels defined using the `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY` attribute, correct?  That doesn't seem right. 

https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_kernels.asciidoc#behavior-with-kernel-bundle-functions-in-the-core-sycl-specification

> The function get_kernel_ids() returns the kernel identifiers for any free function kernels defined by the application, in addition to identifiers for any kernels defined as lambda expressions or named kernel objects.

Reply:
> But this would remove kernels defined using the `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY` attribute, correct?

No, only kernels compiled through `kernel_compiler`, which get a `$`-sign [prefix](https://github.com/intel/llvm/blob/c8a040fb10669d312e4fd413d28f1831894e48c6/sycl/source/detail/jit_compiler.cpp#L1321) prepended to their offload entry's name, which ends up being the `kernel_id`.